panc/src/main/java/org/quattor/pan/dml/data/ProtectedListResource.java

```diff
@@ -25,7 +25,11 @@ public Element duplicate() {
 
 	@Override
 	public Element get(Term key) throws InvalidTermException {
-		return baseList.get(key);
+		final Element value = baseList.get(key);
```

Comment:
@gombasg the copying happens 2 lines lower, with the `.protect`, so unless the value you are trying to retrieve is null, a copy does happen, unless the value itself is a protected resource, in which case the reference is returned.

my main point is the following: with the current code, everytime a procteted  resources is used, this deep-copying occurs, and only because the elements of the protected resource are not/never converted in protected resources themself.